datafusion/expr/src/expr_rewriter/mod.rs

```diff
@@ -399,20 +374,20 @@ mod test {
     }
 
     #[test]
-    #[allow(deprecated)]
+    #[ignore = "investigate if this test is needed"]
```

Comment:
The method which supposed to replace deprecated one is fails this test, investigating if this test makes sense any more

Reply:
I pushed a commit to remove the test to your branch